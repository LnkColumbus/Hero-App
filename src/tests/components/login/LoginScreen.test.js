import React from 'react';
import { mount } from 'enzyme';

import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('Tests in LoginScreeen component', () => {

    const contextValues = {
        dispatch: jest.fn(),
    }

    const historyMock = {
        replace: jest.fn(),
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValues}>
            <LoginScreen history={historyMock} />
        </AuthContext.Provider>
    );
    
    test('should render correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should dispatch and navigate', () => {

        const handleClick = wrapper.find('button').prop('onClick');

        handleClick();

        expect( contextValues.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'Marlon',
            }
        });

        expect( historyMock.replace ).toHaveBeenCalledWith('/');

        localStorage.setItem('lastPath', '/dc');
        handleClick();

        expect( historyMock.replace ).toHaveBeenCalledWith('/dc');


    });
    
    
})
