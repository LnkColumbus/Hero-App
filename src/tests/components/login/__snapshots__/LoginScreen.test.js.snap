// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in LoginScreeen component should render correctly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
